test/custom_task_test.go

```diff
@@ -661,6 +741,9 @@ spec:
           type: %s
         observedGeneration: %v
 `, runName, tc.wantRunConditionReason, tc.wantRunConditionStatus, tc.wantRunConditionType, tc.wantRunObservedGeneration)
+					if tc.wantRetriesStatus != "" {
```

Comment:
This is more of a comment on the previous lines - but getting the status and then appending to a string seems unmaintainable (if we change the status, someone would have to know to update these strings) - is there a reason we cannot just use the go structs here?

Reply:
Oh I'm actually not getting the status but composing the expected status, added some comments, hope this can clear some of the confusion here. 

> if we change the status, someone would have to know to update these strings

I may need more clarification here. Do you mean we change the structure of the status or the value of the fields in status? 

About why do we prefer using YAML, as far as I remember, using YAMLs can give our users examples for how to use Custom Runs. @jerop may have other answers that I don't remember clearly.

Reply:
> I may need more clarification here. Do you mean we change the structure of the status or the value of the fields in status?

Structure of the status field

> About why do we prefer using YAML, as far as I remember, using YAMLs can better instruct our users on how to use Custom Runs. @jerop may have other answers that I don't remember clearly 😅.

Sure - I get that the inputs being yaml make it easier to read the pipeline/run under test , but composing large strings  like the pipelineRun below with 14 args that all have to be specified in exactly the right order makes the test harder to read and maintain

Reply:
Updated as we discussed offline. The basic principle is to make the code more readable, so that it would be easier to debug future test failures.